bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/DeviceHandler.java

```diff
@@ -138,12 +139,43 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     } else if (command instanceof DecimalType) {
                         logger.trace("Received DecimalType Command for Channel {}",
                                 thing.getChannel(channelUID.getId()));
+                        for (String str : com) {
+                            if (str.contains("setCurve")) {
+                                uri = prop.get(str + "Uri");
+                                String value = command.toString();
+                                if (ch.getUID().toString().contains("shift")) {
+                                    // read slope from channel property
+                                    String channelId = ch.getUID().toString().replace("#shift", "#slope");
+                                    Channel c = thing.getChannel(new ChannelUID(channelId));
+                                    if (c != null) {
+                                        Map<String, String> p = c.getProperties();
+                                        param = "{\"slope\":" + p.get("slope") + ", \"shift\":" + value + "}";
+                                        Map<String, String> properties = new HashMap<>();
+                                        properties.putAll(p);
+                                        properties.put("shift", value);
+                                        updateChannelProperties(new ChannelUID(channelId), properties);
```

Comment:
If I read the code correctly you are recreating the channel every time a the binding handles a command for the "shift" or "slope" channels. This is not a good idea. Why are you doing that? Just to store the new value to access it in the other case? If yo it would be much easier to store those two values in a private property of the `DeviceHandler` class.